### ISSUE TITLE
Problem with loadsharing and TP0456?

### ISSUE THREAD

**Issue by notecp** [jhughes1010/weather#12]
Hi,
i stumbled upon this post: https://www.instructables.com/Load-Sharing-Use-Solar-Panel-Safely-With-TP4056/ do we have a prolem here?

Regards,
notecp

**Comment by jhughes1010** [comment]
This is good stuff (in a bad way). I need to think this through and understand What I would want to do with the design. We don't want to shut off the load while the battery is charging, per se. Maybe we need to have a switch for the panel itself possibly, monitor charge cycles in software, etc...

Fortunately, Our load is not a big current consumer (duty cycle for WAKE vs. sleep), but I'll need to investigate.

**Comment by notecp** [comment]
okay.
they even provided a fix for this on the site. Maybe it is worth a try?

**Comment by jhughes1010** [comment]
I don't like the idea of the load (weather station) being turned off every sunny day when the battery is charging. Any daylight hours when battery is needing charge, the station will be down. I'm OK with a FET switch, just not placed there (for our application needs.

I still want to do some research on the full module we are using to understand more before coming up with a proposal.

**Comment by notecp** [comment]
i think it will siwtch automatically around the tp chip when the sun is shining.
the load will stay powered all the time i think

**Comment by jhughes1010** [comment]
I stand corrected. I missed the diode feeding the load directly in the schematic. Yes, we should investigate this.

**Comment by jhughes1010** [comment]
I'm going to look into this some more. It intrigues me...1. Does the TP4056 work normally with a USB power supply plugged in (not a solar panel)? I'm thinking it's more a function of the wimpy solar panel, we end up current starving and VBAT+ never reaches 4.2V. I have a 20V/25W solar panel and a high current buck converter between the panel and the TP4056. 2. Don't we determine the battery is charged with monitoring the voltage at VBAT (again, healthy supply handles the battery and the load without current staring)

I keep trying to rationalize that the circuit will work well when a USB supply is connected. Your thoughts?

**Comment by therocksweather** [comment]
hi James
found this info here
https://www.addicore.com/TP4056-Charger-and-Protection-Module-p/ad310.htm#powerSupplyOptions

Protection Features

This module uses the TP4056 / TC4056 Li-Ion charge controller IC and a
separate protection IC. There are other types of modules on the market that
use the TP4056 / TC4056 but lack any protection circuits or ICs to provide
the necessary protection needed with lithium batteries. This module uses
both the TP4056 / TC4056 and the DW01A Li-Ion battery protection IC, which
together in combination provide the following protection features:

   - Manage the constant current to constant voltage charging of a
   connected lithium battery
   - Over-discharge protection - keeps your battery from being discharged
   below 2.4V, a healthy minimum voltage level for your battery
      If a connected battery has been discharged below 2.4V the module will
      cut output power from the battery until the battery voltage has been
      re-charged above 3.0V (the over-discharge release voltage), which at that
      time the module will again allow discharge of power from the battery to a
      connected load. Although the module cuts output power from the battery
      during an over-discharge situation, it still allows charging of
the battery
      to occur through the parasitic diode of the discharge control MOSFET
      (FS8205A Dual MOSFET).
   - Overcharge protection - the module will safely charge your battery to
   4.2V
   - Overcurrent and short-circuit protection - the module will cut the
   output from the battery if the discharge rate exceeds 3A or if a
   short-circuit condition occurs
   - Soft-start protection limits inrush current
   - Trickle charge (battery reconditioning) - if the voltage level of the
   connected battery is less than 2.9V, the module will use a trickle charge
   current of 130mA until the battery voltage reaches 2.9V, at which point the
   charge current will be linearly increased to the configured charge current.


more info can be found using the resource links ........to me it looks like
the protection IC DW01A already does something similar .........
and similar to u, I am using a larger solar panel into a 3s lithium pack
which then supplies the weather station board via a DC/DC 12/5V board.

I monitor both battery levels, of course the weather station board battery
volts never changes .........but the 3s pack goes up and down ........ as
expected

I think u may be correct in saying .......the original solar panel is a bit
undersized ......maybe .......but what sort of load is it driving?????

remember I am running 1 minute wake intervals and each wake is approx 5 to
7 seconds
u can see the battery voltages here
oxleyweather.ddnsfree.com/telemetry.html

regards Ian

**Comment by notecp** [comment]
i tested it with a powersupply (capable of up to 2.5 A at 6V) attached to the input.
The board finishes the loading and turns blue (while the ESP32 is off via the switch). As soon as i turn on the ESP it immediately starts charging the lithium pack. The power drawn went up from approx 0 mA to about 100 mA (sadly no digital gauge).

my conlusion is, that my linked articel semms to be true and we have to investigate further.

**Comment by jhughes1010** [comment]
I'm not saying that the observations are bad. I actually like the circuit modification. I don't like the idea of immediately trying to top off the battery every time the system turns on and I like the idea of letting the battery actually get the opportunity to discharge.

Are we actually over-charging though? Do we have a fire hazard on our hands? My analogy: An over zealous waiter, waiting to refill your water glass as you pick it up to drink from it. He does not over fill it, but he is there to fill the glass as you drink.

If the battery voltage is not > 4.2V and the system stops charging at 4.2V then are we truly over-charging? Are there any documented cases of this happening where batteries are getting damaged or systems dying, or fires? I don't think the problem is just related to solar charging.

Thinking ahead: I'd also like to include the ability to kill the solar charging unit entirely at low temperatures (inside the case). I'll probably use the same type of FET switch for that also, but drive that FET from the ESP so I can make decisions in software. We could also initiate "full discharge cycle" on the battery periodically, if we desire. This assumes you are using the newer design that does not aggressively discharge the battery nightly.

Ian, transmitter battery voltage does not ever change for you. No sign of discharge, even at night. Am I missing something?

**Comment by jhughes1010** [comment]
I've updated the v4 PCB design goals to comprehend these inputs.

https://jameshughes.atlassian.net/wiki/spaces/WSV4/overview

**Comment by therocksweather** [comment]
Hey James
the circuit is a clever idea, and as u suggested, if driven by a GPIO, then
the user can do all sorts of smart things in software.
The source of this idea though seems to be a bit foggy ...... it indicates
that an overcharge issue is possible if u r using a solar panel.
Or did I miss read something there ????
But a solar panel only works for half a day at best ......and only if there
is sun .......
would it not be more of a concern if u were using a USB charger 24/7 ??????

This is the situation I am in .......my weather station board is connected
to a 5V supply 24/7
hopefully this will explain the setup ...........its not as good as a
picture but

solar panel >>> 18V >>> DC/DC  >>>> 12.6V >>>>> DC/DC >>>> 5V >>>>>>
weather station board
 3S lithium battery pack/charger >>>>>>> ^

solar panel drives (nomial18V) DC/DC (12V6) that charges a 3S lithium pack,
this is also connected to another DC/DC (5V) that is connected to the
weather station board.
basically it is the big battery that fluctuates .... according to the
sun/load .......
only if the big battery is depleted and there is no solar .....will the
weather station board battery start to discharge ......

so if there is an issue with overcharging ...... will I not be there before
a weather station board that uses only a solar panel as a source of supply
?????

So the circuit mod would be useful for me, but only if driven via software,
but the original article hasn't convinced me that there is a problem with
overcharging .....
and certainly not using a solar panel as input supply.

**Comment by therocksweather** [comment]
Hey James
after much digging around I think I have found something .........
this maybe the source of the foggy article ......

https://www.best-microcontroller-projects.com/tp4056.html

 have a read ........ some interesting material .........certainly more
info than the original ......
this may be another case of a manufacturer not clearly stating the purpose
and use of their product ........

**Comment by jhughes1010** [comment]
Ian, great article! I can now get my head around the "why". I did like the conclusion at the end of part 2, LOL. Nobody's ever had a problem with it, but not the right thing to do. I'm convinced that it is the right path to take.

My take on the solution to the problem. I'm envisioning a "pass through" PCB that sits below the 4056 module. This could be a useful solution for many projects. We can keep it off the weather station PCB altogether. It would be available for any project though. The only "hack" would be connecting the TEMP pin to the lower PCB (next paragraph).

I also found it interesting that the 4056 has a pin for TEMP control, but it states that the NTC resistor is part of the battery. Users will not want to purchase a more expensive battery, so we would take care of the low temp case and high temp case by tacking a thermistor onto the battery? No, we are not measuring the temperature in the battery itself, but can still control high/low temperature settings on the charger circuit itself.

On to your specific application: How do you manage the charging of your 3S LiPo pack? You are not loading a battery into the weather station PCB at all, right?

**Comment by therocksweather** [comment]
😀 yeah .....I read the same conclusion ........ but I guess I don't want 2
b the first on the stats list!!!!!

the 3S pack is one of these
https://www.ebay.com.au/itm/284154175427

sadly , there is very little info about it compared to the TP4056 ......
but it appears to work ........no complaints ..........yet
the charge/discharge connections are one of the same ...... I treat it just
like a battery ....
a bit like your car battery .......charger is connected ......the load is
connected ...... and it just works .....
if they put a handful of electronics inside your battery case ....... would
u b any the wiser?????

no the weather station has its battery in place.
so consider the weather station to be one of several loads supplied by the
3S pack (but the most important)

**Comment by jhughes1010** [comment]
Very raw and unchecked at the moment, but I'm thinking of something like this for the load switching...
<img width="457" alt="image" src="https://user-images.githubusercontent.com/585299/167465420-0a0cf3f2-8b1d-4f6a-9268-52ee6b468fe0.png">

**Comment by jhughes1010** [comment]
Ok, doing final checkout of PCB before fabrication, but here's the 3D view.
<img width="1879" alt="image" src="https://user-images.githubusercontent.com/585299/168801725-533d5042-26d6-4eba-9f31-c8b73146e6f6.png">

**Comment by jhughes1010** [comment]
Gerbers have been created and PCB is in fab now.
<img width="1585" alt="image" src="https://user-images.githubusercontent.com/585299/169264754-68434639-7a19-4326-9beb-5c66429f1864.png">

**Comment by jhughes1010** [comment]
FYI, this retro-fit PCB may be a tight fit to the weather station PCB. You may consider mounting it from the underside of the PCB. I'll be making the gerbers available as soon as I receive and verify the design.

**Comment by Thijs74** [comment]
Hi James,
Interesting thoughts, earlier today I stumbled across a video handling the same issue. The schematics are quite similar, and well explained. https://www.youtube.com/watch?v=37kGva3NW8w&t=32s

**Comment by jhughes1010** [comment]
Thanks for sharing the post! I respect Andreas work and publishing. I think this confirms we are on a good course. PCBs have cleared US customs and headed to Denver, CO today.

**Comment by therocksweather** [comment]
Hey James
after much digging around I think I have found something .........
this maybe the source of the foggy article ......

https://www.best-microcontroller-projects.com/tp4056.html

 have a read ........ some interesting material .........certainly more
info than the original ......
this may be another case of a manufacturer not clearly stating the purpose
and use of their product ........
